app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt

```diff
@@ -28,13 +29,19 @@ fun TopTabs(tripsViewModel: TripsViewModel, navigationActions: NavigationActions
   // Remember the currently selected tab index
   var selectedTabIndex by remember { mutableIntStateOf(0) }
 
-  val trip =
-      tripsViewModel.selectedTrip.value
-          ?: return Text(text = "No trip selected. Should not happen", color = Color.Red)
+  // Collect selectedTrip as state to avoid calling .value directly in composition
+  val trip by tripsViewModel.selectedTrip.collectAsState()
+
+  // Check if the selected trip is null and navigate to "Overview" if true
+  if (trip == null) {
+    navigationActions.navigateTo(Route.OVERVIEW)
+    return
+  }
 
   // Column for top tabs and content
   Column(modifier = Modifier.testTag("topTabs")) {
-    TopBarWithImage(trip, navigationActions)
+    TopBarWithImage(trip!!, navigationActions)
```

Comment:
instead of using trip!! everywhere let's do something like 
val selectedTrip = trip!! just once.

Reply:
Thank you for your suggestion! I've implemented the changes as recommended.